backend/fms_core/template_importer/importers/_generic.py

```diff
@@ -0,0 +1,108 @@
+from pandas import pandas as pd
+from django.db import transaction
+import reversion
+
+from ..sheet_data import SheetData
+from .._utils import blank_and_nan_to_none
+
+class GenericImporter():
+    ERRORS_CUTOFF = 20
+
+    def __init__(self):
+        self.base_errors = []
+        self.errors_count = 0
+
+        self.preloaded_data = {}
+        self.file = None
+        self.sheets = {}
+        self.previews_info = []
+
+        print('Generic Importer - __init__ successful')
+
+
+    def import_template(self, file, format, dry_run):
+        self.file = file
+
+        print('file', file)
+
+        for sheet_info in self.SHEETS_INFO:
+            sheet_name = sheet_info['name']
+            print('sheet_name', sheet_name)
+            sheet_created = self.create_sheet_data(sheet_name=sheet_name, header_row_nb=sheet_info['header_row_nb'])
+
+            if sheet_created is not None:
+                self.sheets[sheet_name] = sheet_created
+
+        print('sheets')
+
+        if not self.base_errors:
+            with transaction.atomic():
+                if dry_run:
+                    # This ensures that only one reversion is created, and is rollbacked in a dry_run
+                    with reversion.create_revision(manage_manually=True):
+                        self.import_template_inner()
+                        reversion.set_comment("Template import - dry run")
+                    transaction.set_rollback(True)
+                else:
+                    self.import_template_inner()
+                    reversion.set_comment("Template import")
+
+            # Add processed rows with errors/warnings/diffs to self.previews_info list
+            for sheet in list(self.sheets.values()):
+                preview_info = sheet.generate_preview_info_from_rows_results(rows_results=sheet.rows_results)
+                self.previews_info.append(preview_info)
+
+        has_warnings = False
+        for sheet_preview in self.previews_info:
+            if any([r['warnings'] for r in sheet_preview['rows']]):
+                has_warnings = True
+                break
+
+        import_result = {'valid': self.is_valid,
+                         'has_warnings': has_warnings,
+                         'base_errors': [{
+                             "error": str(e),
+                             } for e in self.base_errors],
+                         'result_previews': self.previews_info,
+                         }
+        return import_result
+
+
+    def create_sheet_data(self, sheet_name, header_row_nb):
+        try:
+            pd_sheet = pd.read_excel(self.file, sheet_name=sheet_name)
+            # Convert blank and NaN cells to None and Store it in self.sheets
+            dataframe = pd_sheet.applymap(lambda x: blank_and_nan_to_none(x))
```

Comment:
pretty confident that this is the same as dataframe = pd_sheet.applymap(blank_and_nan_to_none). Not sure i have ever found a good reason to use lambda in python ... (or anywhere other than lisp ;) this is like wrapping your function in a function ;)